Keas.Mvc/Controllers/EquipmentAttributeKeyController.cs

```diff
@@ -0,0 +1,155 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Keas.Core.Data;
+using Keas.Core.Domain;
+using Keas.Core.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace Keas.Mvc.Controllers
+{
+    [Authorize(Policy = AccessCodes.Codes.DepartmentAdminAccess)] //TODO: Decide if we want equip manager to have access
+    public class EquipmentAttributeKeyController : SuperController
+    {
+        private readonly ApplicationDbContext _context;
+
+        public EquipmentAttributeKeyController(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<IActionResult> Index()
+        {
+            var model = await _context.EquipmentAttributeKeys.Where(a => a.TeamId == null || a.Team.Slug == Team).ToListAsync();
+            return View(model);
+        }
+
+        public IActionResult Create()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Create(EquipmentAttributeKey model)
+        {
+            var team = await _context.Teams.FirstAsync(t => t.Slug == Team);
+
+            if (ModelState.IsValid)
+            {
+                if (await _context.EquipmentAttributeKeys.AnyAsync(a => a.TeamId == null && a.Key.Equals(model.Key.Trim(), StringComparison.OrdinalIgnoreCase)))
+                {
+                    ModelState.AddModelError("Key", "This Key already exists as a global key.");
+                }
+                else
+                {
+                    if (await _context.EquipmentAttributeKeys.AnyAsync(a => a.TeamId == team.Id && a.Key.Equals(model.Key.Trim(), StringComparison.OrdinalIgnoreCase)))
+                    {
+                        ModelState.AddModelError("Key", "This Key already exists.");
+                    }
+                }
+            }
+
+            if (ModelState.IsValid)
+            {
+                if (team == null)
+                {
+                    throw new Exception("This should never happen.");

```

Comment:
what's the point?  we know team is not null because of the `firstAsync()` call above

Reply:
@srkirkland Because if the code ever got changed, a team could add to the global values.